### PR TITLE
Issue31 opt codes

### LINKED ISSUE #31: reduce parsing time
This issue intends to optimize java codes, to reduce parsing time.

### PR DESCRIPTION
**Pull request by JayHuLBL**
This closes #31.

### CODE REVIEW

File: java/parser/src/main/java/gov/lbl/parser/domain/Declaration.java [hunk 1]
```diff
@@ -20,32 +22,27 @@ public Declaration(String ident,
...
+    				temStrR = modification.substring(equSymbol.get(0)+1,modification.length());
+    			} else {
+    				temStrL = modification.substring(1,modification.length()-1);
     			}
-    			ClassMod classMod = new ClassMod();
-    			classMod.classMod(tempClaStr);
-    			this.class_modification = classMod;
+        		if (temStrL.charAt(temStrL.length()-1) == ' ') {
+        			temStrL = temStrL.substring(0, temStrL.length()-1);
+        		}
```
Comment (mwetter) [inline java/parser/src/main/java/gov/lbl/parser/domain/Declaration.java]: Is there are reason why you are not simply calling the trim() method on the string?

File: java/parser/src/main/java/gov/lbl/parser/domain/Declaration.java [hunk 2]
```diff
@@ -199,8 +77,13 @@ private Mod classMod(String classModStr) {
         		PerMod per_modification = null;
     			if (!strSets.get(i).contains("(")
     					|| (strSets.get(i).contains("(") && (strSets.get(i).indexOf('(')> strSets.get(i).indexOf('=')))) {
-    				tempLeftStr = strSets.get(i).split("=")[0];
-    				tempRightStr = strSets.get(i).split("=")[1];
+    				if (strSets.get(i).contains("=")) {
+    					tempLeftStr = strSets.get(i).split("=")[0];
```
Comment (mwetter) [inline java/parser/src/main/java/gov/lbl/parser/domain/Declaration.java]: you should call split("=") only once, not twice, as this is expensive.

File: java/parser/src/main/java/gov/lbl/parser/domain/Declaration.java [hunk 3]
```diff
@@ -216,24 +99,42 @@ private Mod classMod(String classModStr) {
...
-    				value = tempRightStr;
-    			} else if ((strSets.get(i).contains("(") && (strSets.get(i).indexOf('(')< strSets.get(i).indexOf('=')))) {
+    				value = tempRightStr.isEmpty() ? null : tempRightStr;
+    			} else if ((strSets.get(i).contains("(") && (strSets.get(i).indexOf('(')< strSets.get(i).indexOf('=')))) {    				
     				String variable = strSets.get(i).substring(0,strSets.get(i).indexOf('(')-1);
-    				if (!variable.equals("per")) {
+    				String[] temStr = variable.split(" ");
+    				if (temStr.length > 1) {
+    					String prefixTemp = "";
+    					for (int j=0; j<temStr.length-1; j++) {
```
Comment (mwetter) [inline java/parser/src/main/java/gov/lbl/parser/domain/Declaration.java]: Use the `join(...)` method of Java rather than a loop.

**Comment by JayHuLBL** [comment]
@mwetter Thanks for your comments. These are very helpful to further simplify the code. I just updated the codes according to your comments.

**Comment by JayHuLBL** [comment]
@mwetter Please let me know if it is appropriate now. Sorry for the back and forth.

**Pull request merged**